main/src/ca/uwaterloo/flix/language/phase/CodegenHelper.scala

```diff
@@ -147,6 +147,91 @@ object CodegenHelper {
   }
 
   /**
+    * This case class holds all the information required for an Enum case
+    * @param qualName Qualified Name of the cas
+    * @param executableCase Representation in the AST
+    * @param fieldType Type of the field of the case
+    */
+  case class EnumCaseInfo(qualName: EnumClassName, executableCase: ExecutableAst.Case, fieldType: Type)
+
+  /**
+    * Generates that holds all information required for enums
+    * @param root Root of the executable AST
+    */
+  def generateEnumTypeInfoMap(root: ExecutableAst.Root): Map[Type, Map[String, EnumCaseInfo]] = {
+    // This list contains all enum cases with their type and with the type of their field
+    val allEnums: List[(Type, (String, Type))] = root.defs.values.flatMap(x => CodegenHelper.findEnumCases(x.exp)).toList
```

Comment:
We need to refactor code like this to be maintainable. (We will talk about how to do this.)

Reply:
We should talk about this but this is the best I can do in the backend with what I have.